### PR TITLE
Automatic port of System.Globalization.CompareInfo docs

### PR DESCRIPTION
**Pull request by carlossanlop**
Ported with [DocsPortingTool](https://github.com/carlossanlop/DocsPortingTool).

### CODE REVIEW

File: xml/System.Globalization/CompareInfo.xml [hunk 1]
```diff
@@ -2222,13 +2222,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
       </Parameters>
       <Docs>
-        <param name="source">To be added.</param>
-        <param name="value">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="matchLength">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The string to search within.</param>
```
Comment (tarekgh) [inline xml/System.Globalization/CompareInfo.xml:2225]: >string [](start = 33, length = 6)

didn't we change that to span in other APIs?

File: xml/System.Globalization/CompareInfo.xml [hunk 2]
```diff
@@ -2222,13 +2222,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <param name="value">The substring to locate within <paramref name="source" />.</param>
+        <param name="options">The <see cref="T:System.Globalization.CompareOptions" /> to use during the search.</param>
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired value. This may be different than the length of <paramref name="value" /> if a linguistic comparison is performed. Set to 0 if <paramref name="value" /> is not found within <paramref name="source" />.</param>
+        <summary>Searches for the first occurrence of a substring within a source string.</summary>
+        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> first appears; or -1 if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
```
Comment (tarekgh) [inline xml/System.Globalization/CompareInfo.xml:2230]: >-1 [](start = 134, length = 2)

maybe call this negative value instead.

File: xml/System.Globalization/CompareInfo.xml [hunk 3]
```diff
@@ -4218,13 +4250,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
-        <param name="matchLength">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The string to search within.</param>
+        <param name="value">The substring to locate within <paramref name="source" />.</param>
+        <param name="options">The <see cref="T:System.Globalization.CompareOptions" /> to use during the search.</param>
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired value. This may be different than the length of <paramref name="value" /> if a linguistic comparison is performed. Set to 0 if <paramref name="value" /> is not found within <paramref name="source" />.</param>
+        <summary>Searches for the last occurrence of a substring within a source string.</summary>
+        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> last appears; or -1 if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
```
Comment (tarekgh) [inline xml/System.Globalization/CompareInfo.xml:4258]: >-1  [](start = 133, length = 3)

ditto

File: xml/System.Globalization/CompareInfo.xml [hunk 4]
```diff
@@ -4218,13 +4250,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method has greater overhead than other <xref:System.Globalization.CompareInfo.IndexOf(System.String,System.Char)> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```
Comment (tarekgh) [inline xml/System.Globalization/CompareInfo.xml:4264]: >IndexOf [](start = 83, length = 7)

This should be LastIndexOf

File: xml/System.Globalization/CompareInfo.xml [hunk 5]
```diff
@@ -2222,13 +2222,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <param name="options">The <see cref="T:System.Globalization.CompareOptions" /> to use during the search.</param>
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired value. This may be different than the length of <paramref name="value" /> if a linguistic comparison is performed. Set to 0 if <paramref name="value" /> is not found within <paramref name="source" />.</param>
+        <summary>Searches for the first occurrence of a substring within a source string.</summary>
+        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> first appears; or -1 if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method has greater overhead than other <xref:System.Globalization.CompareInfo.IndexOf(System.String,System.Char)> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```
Comment (tarekgh) [inline xml/System.Globalization/CompareInfo.xml:2236]: >System.String,System.Char [](start = 91, length = 25)

don't specify parameters here. overloads can have different parameters.

File: xml/System.Globalization/CompareInfo.xml (hunk 4 above)
Comment (tarekgh) [inline xml/System.Globalization/CompareInfo.xml:4264]: >System.String,System.Char [](start = 91, length = 25)

don't specify parameters here. overloads can have different parameters.

File: xml/System.Globalization/CompareInfo.xml [hunk 6]
```diff
@@ -2222,13 +2222,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
       <Docs>
-        <param name="source">To be added.</param>
-        <param name="value">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="matchLength">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The string to search within.</param>
+        <param name="value">The substring to locate within <paramref name="source" />.</param>
```
Comment (carlossanlop) [inline xml/System.Globalization/CompareInfo.xml:2226]:
```suggestion
        <param name="source">The read-only span of characters to search within.</param>
        <param name="value">The read-only span of characters containing the substring to locate within <paramref name="source" />.</param>
```

File: xml/System.Globalization/CompareInfo.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Globalization/CompareInfo.xml:2230]:
```suggestion
        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> first appears; or a negative value if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
```

File: xml/System.Globalization/CompareInfo.xml (hunk 5 above)
Comment (carlossanlop) [reply xml/System.Globalization/CompareInfo.xml:2236]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.IndexOf%2A> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

File: xml/System.Globalization/CompareInfo.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Globalization/CompareInfo.xml:4258]:
```suggestion
        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> last appears; or a negative number if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
```

File: xml/System.Globalization/CompareInfo.xml (hunk 4 above)
Comment (carlossanlop) [reply xml/System.Globalization/CompareInfo.xml:4264]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.LastIndexOf%2A> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

Comment (carlossanlop) [reply xml/System.Globalization/CompareInfo.xml:4264]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.IndexOf%2A> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

File: xml/System.Globalization/CompareInfo.xml [hunk 7]
```diff
@@ -3097,13 +3107,24 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
       <Docs>
-        <param name="source">To be added.</param>
-        <param name="prefix">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="matchLength">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The string to search within.</param>
+        <param name="prefix">The prefix to attempt to match at the start of <paramref name="source" />.</param>
```
Comment (carlossanlop) [inline xml/System.Globalization/CompareInfo.xml:3111]:
```suggestion
        <param name="source">The read-only span of characters to search within.</param>
        <param name="prefix">The read-only span of characters containing the prefix to attempt to match at the start of <paramref name="source" />.</param>
```

File: xml/System.Globalization/CompareInfo.xml [hunk 8]
```diff
@@ -3556,13 +3577,24 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
       <Docs>
-        <param name="source">To be added.</param>
-        <param name="suffix">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="matchLength">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The string to search within.</param>
+        <param name="suffix">The suffix to attempt to match at the end of <paramref name="source" />.</param>
```
Comment (carlossanlop) [inline xml/System.Globalization/CompareInfo.xml:3581]:
```suggestion
        <param name="source">The read-only span of characters to search within.</param>
        <param name="suffix">The read-only span of characters containing the suffix to attempt to match at the end of <paramref name="source" />.</param>
```

File: xml/System.Globalization/CompareInfo.xml [hunk 9]
```diff
@@ -4218,13 +4250,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
       <Docs>
-        <param name="source">To be added.</param>
-        <param name="value">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="matchLength">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="source">The string to search within.</param>
+        <param name="value">The substring to locate within <paramref name="source" />.</param>
```
Comment (carlossanlop) [inline xml/System.Globalization/CompareInfo.xml:4254]:
```suggestion
        <param name="source">The read-only span of characters to search within.</param>
        <param name="value">The read-only span of characters containing the substring to locate within <paramref name="source" />.</param>
```

**Review by tarekgh — APPROVED** [review]

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**